test/general/buffer.jl

```diff
@@ -0,0 +1,67 @@
+@testset verbose=true "SystemBuffer" begin
+    # Mock fluid system
+    struct FluidSystemMock3 <: TrixiParticles.FluidSystem{2, Nothing} end
+
+    inflow = InFlow(; plane=([0.0, 0.0], [0.0, 1.0]), particle_spacing=0.2,
+                    open_boundary_layers=2, density=1.0, flow_direction=[1.0, 0.0])
+    system = OpenBoundarySPHSystem(inflow; sound_speed=1.0, fluid_system=FluidSystemMock3(),
+                                   buffer_size=0)
+    system_buffer = OpenBoundarySPHSystem(inflow; sound_speed=1.0, buffer_size=5,
+                                          fluid_system=FluidSystemMock3())
+
+    n_particles = nparticles(system)
+
+    @testset "Iterators" begin
+        @test Base.OneTo(n_particles) == TrixiParticles.each_moving_particle(system)
```

Comment:
```suggestion
        @test TrixiParticles.each_moving_particle(system) == 1:n_particles
```
You can just use `1:n` here and below.